docs/rule/no-mut-helper.md

```diff
@@ -0,0 +1,244 @@
+
+Example:
+
+```hbs
+<input
+  type="checkbox"
+  checked={{@checked}}
+  {{on "change" (fn (mut @checked) (not @checked))}}
+/>
+```
+
+## What this rule does
+
+This rule forbids any use of the `mut` helper, both as a getter and a setter, in any context. It also
+surfaces possible alternatives in the lint violation message to help guide engineers to resolving
+the lint violations.
+
+## Examples
+
+This rule **forbids** the following:
+
+```hbs
+<MyComponent @isDropdownOpen={{mut this.isDropdownOpen}}/>
+```
+
+```hbs
+<button {{action (mut this.isDropdownOpen) false}}>Close Dropdown</button>
+```
+
+```hbs
+<MyComponent @closeDropdown={{action (mut this.isDropdownOpen) false}}/>
+```
+
+```hbs
+<MyComponent onclick={{fn (mut this.isDropdownOpen) false}}/>
+```
+
+```hbs
+<Input onchange={{action (mut this.profile.description) value="target.value"}}/>
+```
+
+```hbs
+{{my-component click=(action (mut this.isDropdownOpen) false)}}
+```
+
+```hbs
+{{my-component value=(mut this.profile.description)}}
+```
+
+This rule **allows** the following:
+
+```hbs
+<MyComponent @isDropdownOpen={{this.isDropdownOpen}}/>
+```
+
+```hbs
+<button {{action (ember-set-helper$set this "isDropdownOpen" false)}}>Close Dropdown</button>
+```
+
+```hbs
+<button {{on "click" (ember-set-helper$set this "isDropdownOpen" false)}}>Close Dropdown</button>
+```
+
+```hbs
+<button {{on "click" (fn this.setIsDropdownOpen false)}}>Close Dropdown</button>
+```
+
+```hbs
+<button {{action "setIsDropdownOpen" false}}>Close Dropdown</button>
+```
+
+```hbs
+<MyComponent @closeDropdown={{action this.setIsDropdownOpen false}}/>
+```
+
+```hbs
+<MyComponent onclick={{fn this.setIsDropdownOpen false}}/>
+```
+
+```hbs
+<Input onchange={{action this.setProfileDescription}}/>
+```
+
+Coupled with a corresponding JS action to set:
+
+```js
+@action
+setProfileDescription({ target: { value } }) {
+  set(this, 'profile.description', value);
+}
+```
+
+```hbs
+{{my-component click=(action (ember-set-helper$set this "setIsDropdownOpen" false)}}
+```
+
+```hbs
+{{my-component value=this.profile.description}}
+```
+
+## Migration
+
+1. When used as a pure setter only, `mut` could be replaced by a JS action or [ember-set-helper](https://github.com/pzuraq/ember-set-helper):
+
+Before:
+
+```hbs
+<MyComponent
+  @closeDropdown={{action (mut this.setIsDropdownOpen) false}}
+/>
+```
+
+After (Option 1 HBS):
+
+```hbs
+<MyComponent
+  @closeDropdown={{action this.setIsDropdownOpen false}}
+/>
+```
+
+After (Option 1 JS):
+
+```js
+@action
+setIsDropdownOpen(isDropdownOpen) {
+  set(this, 'isDropdownOpen', isDropdownOpen);
+}
+```
+
+After (Option 2):
+
+```hbs
+<MyComponent
+  @closeDropdown={{ember-set-helper$set this "isDropdownOpen" false}}
```

Comment:
`$set` looks redundant

Reply:
fair point, Alex. I have updated the Pull Request to no longer use the addon name prefix when invoking the `{{set}}` helper.

fixed - thanks for the suggestion.